### DISCUSSION TITLE
Is it possible to make use of "world info" or "memory" via the API?

### DISCUSSION THREAD

**Discussion by VL4DST3R** [LostRuins/koboldcpp#1139] [Q&A] [answered]
Basically the title: Is it possible to make use of context data when generating via the API? Where it will dynamically load and append world info "memories" when a keywords gets mentioned? Or you would have to essentially replicate the functionality locally? (find valid keywords present in the submitted text and append the relevant info associated with it)

**Comment by LostRuins** [comment]
Kobold lite has this feature but not available directly over the api

**Reply by VL4DST3R** [reply]
I see, and is this planned for the future?

**Reply by LostRuins** [reply, marked as answer]
not over the api.

**Reply by VL4DST3R** [reply]
I understand, welp, I guess I will have to hack something together myself then. Can you by chance give me a very quick overview of how you achieved it in the lite ui? Do you just look for any matching keywords and append the info to the prompt or is there some other secret sauce?

**Reply by LostRuins** [reply]
Yes, that's the easiest way to do it. Match by keyword, and then add the value to context.

This may help

https://github.com/LostRuins/ConcedoBot

**Reply by VL4DST3R** [reply]
I've already checked out your bot, as this is what I'm currently trying to do for me and my friends as well XD But thanks for the info.
